docs/posts/ibis-analytics/index.qmd

```diff
@@ -1,341 +1,1969 @@
+Ibis (and co-created Apache Arrow) to address these issues and more with a
+different approach than any other Python dataframe library by decoupling the
+dataframe API from the backend implementation. This allows Ibis to support 20+
+backends today, including pandas and DuckDB and Polars and Postgres and BigQuery
+and Snowflake and PySpark and over a dozen more.
+
+I've experienced many pains of the ecosystem myself. I've downloaded a 1 GB CSV
+from a database GUI and used pandas to munge it because I couldn't figure it out
+in SQL. When building larger ML demos that included data preprocessing on GBs, I
+hit OOM errors and learned about how much data expands in memory and why to
+store files in compressed Parquet. I learned about partitioning strategies and
+the joy of subtle differences and incompatibilities across data system. I
+started using PySpark and Dask to overcome hurdles. I slowly learned about
+databases. Fortunately while I've been learning, others ([including many Ibis
+contributors](https://github.com/ibis-project/ibis/contributors)) have been
+building a modern, open-source Python data ecosystem! Now I can just use what
+they've built.
+
+In this blog, we'll look at how to build an end-to-end analytics project using
+Ibis as the frontend for data in Python. Ibis plays a large part, but is only
+one component in any real data project.
+
+We'll combine a variety of open-source tools and freemium services including:
 
 - [Ibis](https://ibis-project.org) (dataframe library)
 - [DuckDB](https://duckdb.org) (database and query engine)
+- [Dagster](https://dagster.io) (orchestration library)
 - [Plotly](https://plotly.com/python) (visualization library)
 - [Streamlit](https://streamlit.io) (dashboard library)
-- [justfile](https://just.systems/man) (command runner)
+- [justfile](https://just.systems) (command runner)
 - [TOML](https://toml.io) (configuration)
 - [GitHub](https://github.com) (source control, CI/CD)
+- [Azure VM](https://azure.microsoft.com/products/virtual-machines) (self-hosted runner)
 - [GitHub web APIs](https://docs.github.com/en/graphql) (source data)
 - [Google BigQuery](https://cloud.google.com/free/docs/free-cloud-features#bigquery) (source data)
-- [Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs) (cloud storage backup)
+- [Zulip](https://zulip.com) (source data)
 - [MotherDuck](https://motherduck.com) (cloud service for DuckDB)
 - [Streamlit Community Cloud](https://docs.streamlit.io/streamlit-community-cloud) (cloud service for Streamlit)
 
-to build an open-source, modular, composable, scalable, automated, hybrid-cloud, end-to-end analytics project processing a few million rows of data that provides real business value at little to no cost.
+to build an open-source, modular, composable, scalable, automated, hybrid-cloud,
+end-to-end analytics project processing a few million rows of data that provides
+real business value at little to no cost.
```

Comment:
```suggestion
real business value.
```

Probably overpromising a bit here. I think "real business value" is enough justification.